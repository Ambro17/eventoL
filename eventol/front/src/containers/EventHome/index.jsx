import React from 'react';
import PropTypes from 'prop-types';
import withSizes from 'react-sizes';

import Hero from '../../components/Hero';
import Header from '../../components/Header';
import Search from '../../components/Search';
import TitleList from '../../components/TitleList';
import {
  MOBILE_WIDTH,
  BACKGROUND_DEFAULT,
  LOGO_HEADER_DEFAULT,
  LOGO_LANDING_DEFAULT,
  HOME_REQUIRED_FIELDS,
} from '../../utils/constants';

import './index.scss';


class EventHome extends React.Component {
  static propTypes = {
    isMobile: PropTypes.bool.isRequired,
    background: PropTypes.string,
    logoHeader: PropTypes.string,
    logoLanding: PropTypes.string,
    tagMessage: PropTypes.string,
    tagSlug: PropTypes.string,
    user: PropTypes.shape({
      /* eslint-disable camelcase */
      first_name: PropTypes.string,
      last_name: PropTypes.string,
      /* eslint-enable camelcase */
    }),
  }

  static defaultProps = {
    background: BACKGROUND_DEFAULT,
    logoHeader: LOGO_HEADER_DEFAULT,
    logoLanding: LOGO_LANDING_DEFAULT,
    tagMessage: null,
    tagSlug: null,
    user: null,
  }

  state = {
    searchTerm: '',
    searchUrl: '',
    searched: false,
  }

  handleOnEnter = () => {
    const {tagSlug} = this.props;
    const {searchTerm} = this.state;
    if (searchTerm !== ''){
      const searchUrl = `?search=${searchTerm}&tags__slug=${tagSlug}&fields=${HOME_REQUIRED_FIELDS}`; // TODO: move to utils
      this.setState({searchUrl, searched: true});
    }
  }

  handleOnChange = searchTerm => this.setState({searchTerm})

  render(){
    const {searched, searchUrl} = this.state;
    const {
      user, tagSlug, background,
      logoHeader, logoLanding,
      tagMessage, isMobile,
    } = this.props;
    return (
      <div>
        <Header logoHeader={logoHeader} user={user} isMobile={isMobile} />
        <Hero background={background} logoLanding={logoLanding} message={tagMessage} slug={tagSlug}>
          <Search onChange={this.handleOnChange} onEnter={this.handleOnEnter} />
        </Hero>
        {searched && <TitleList showEmpty title={gettext('Search results')} url={searchUrl} />}
        <TitleList
          id='my_events'
          title={gettext('My Events')}
          url={`?my_events=true&tags__slug=${tagSlug}&fields=${HOME_REQUIRED_FIELDS}` /* TODO: move to utils */}
        />
        <TitleList
          id='next'
          title={gettext('Upcoming Events')}
          url={`?registration_is_open=true&ordering=last_date&tags__slug=${tagSlug}&fields=${HOME_REQUIRED_FIELDS}` /* TODO: move to utils */}
        />
        <TitleList
          id='finished'
          title={gettext('Finished Events')}
          url={`?registration_is_open=false&ordering=-attendees_count&tags__slug=${tagSlug}&fields=${HOME_REQUIRED_FIELDS}` /* TODO: move to utils */}
        />
      </div>
    );
  }
}

const mapSizesToProps = ({width}) => ({
  isMobile: width < MOBILE_WIDTH,
});

export default withSizes(mapSizesToProps)(EventHome);
