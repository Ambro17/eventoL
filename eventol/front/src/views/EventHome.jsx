import React from 'react'
import {render} from 'react-dom'
import EventHome from '../containers/EventHome'
import {Provider} from 'react-redux'
import eventsStore from '../store/eventsStore'
import './index.scss';

const store = eventsStore();
window.render_components = properties => {
  window.params = {...properties};
  render(
    (<Provider store={store}>
      <EventHome
        background={properties.background}
        logoHeader={properties.logo_header}
        logoLanding={properties.logo_landing}
        tagMessage={properties.message}
        tagSlug={properties.tag_slug}
        user={properties.user}
      />
    </Provider>), document.getElementById('root'));
};

if (module.hot) {
  if (window.params) window.render_components(window.params);
  module.hot.accept();
}
