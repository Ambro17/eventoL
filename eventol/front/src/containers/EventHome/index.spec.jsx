import React from 'react';
import renderer from 'react-test-renderer';

import EventHome from '.';

describe('EventHome', () => {
  let component, tree, background,
    logoHeader, logoLanding, tagMessage,
    tagSlug, user;

  beforeEach(() => {
    background = '';
    logoHeader = '';
    logoLanding = '';
    tagMessage = '';
    tagSlug = '';
    user = {};
    component = renderer.create(
      <EventHome
        background={background}
        logoHeader={logoHeader}
        logoLanding={logoLanding}
        tagMessage={tagMessage}
        tagSlug={tagSlug}
        user={user}
      />,
    );
    tree = component.toJSON();
  });

  test('Snapshot', () => {
    expect(tree).toMatchSnapshot();
  });
});
