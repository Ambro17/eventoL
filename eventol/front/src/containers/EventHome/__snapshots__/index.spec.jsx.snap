// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventHome Snapshot 1`] = `
<div>
  <header
    className="header"
  >
    <div
      className="nav-wide"
    >
      <div
        className="logo-component"
        id="logo"
      >
        <a
          href="/"
        >
          <img
            src=""
          />
        </a>
      </div>
      <div
        className="navigation"
        id="navigation"
      >
        <nav>
          <ul>
            <a
              onClick={[Function]}
              onKeyPress={[Function]}
              role="link"
              tabIndex="0"
            >
              <li>
                Search
              </li>
            </a>
            <a
              href="http://eventol.github.io/eventoL/#/"
            >
              <li>
                Documentation
              </li>
            </a>
            <a
              href="#featured"
            >
              <li>
                Featured
              </li>
            </a>
            <a
              href="#recent"
            >
              <li>
                Recent
              </li>
            </a>
            <a
              href="/report"
            >
              <li>
                Generic report
              </li>
            </a>
          </ul>
        </nav>
      </div>
      <div
        className="user-profile"
      >
        <a
          href="/accounts/profile/"
        >
          <div
            className="user"
          >
            <div
              className="name"
            >
              undefined undefined
            </div>
            <div
              className="image"
            >
              <div
                className="fa fa-user fa-2x"
              />
            </div>
          </div>
        </a>
      </div>
    </div>
  </header>
  <div
    className="hero"
    id="hero"
    style={
      Object {
        "backgroundImage": "url()",
      }
    }
  >
    <div
      className="content"
    >
      <p>
        <img
          alt="logo"
          className="logo"
          src=""
        />
      </p>
      <h2>
        Search your event
      </h2>
      <div
        className="search"
        id="search"
      >
        <input
          id="search-input"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="Search by title..."
          type="search"
          value=""
        />
      </div>
    </div>
    <div
      className="overlay"
    />
  </div>
</div>
`;
